### PR TITLE
Next

### PR DESCRIPTION
**Pull request by jorgenbuilder**
I'm adding some new game metadata and query methods based on frontend needs.

### CODE REVIEW

**Review by di-wu — COMMENTED** [review]
@jorgenbuilder, can I have another (little longer) description for these functions and their behaviour?

- `getMetascores`
- `getPercentileMetascore`
- `getScoreCount`

**Comment by jorgenbuilder** [comment]
Whatever you need, camper Quint ⛺️🪵

 `getMetascores` with params `offset` and `count` returns a subset of overall metascores from a sorted list. I would have implemented this, but it seems overall score calculations are done on the fly, so no such sorted list exists. It basically does the same thing as the `getGameScores` method that I implemented, but for overall scores.

`getPercentileMetascore` with param `percentile` (0-1 float) returns the overall metascore at the given percentile. For some colour, the frontend would be using this to render the "next rank" panel. Basically, get the overall metascore from sorted list at `overallMetascoresSortedList.size() * percentile float`. Again, relies on a sorted list of overall metascores in memory.

`getScoreCount` simply returns the number of unique scores in memory (overall Metascores included or not, doesn't really matter.) There is data in memory to write this method against, but I think it would be smarter to initialize a `var scoreCount : Nat` so that we aren't traversing our entire memory. This method powers a vanity metric in the footer of the app. It's a nice-to-have, not essential; and perhaps I'm overthinking/wrong about the inefficiency 🤔

File: metascore/AccountRecord.mo [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import Nat "mo:base/Nat";
+import Nat32 "mo:base/Nat32";
+
+module {
+
+    public type AccountId = Nat;
+
+    // A user account.
+    public type AccountRecord = {
+        id : Nat;
```
Comment (di-wu) [inline metascore/AccountRecord.mo:12]: We should use the `AccountId` type here.

**Review by di-wu — COMMENTED** [review]
✨

**Pull request closed without merging**